sdks/java/harness/src/main/java/org/apache/beam/fn/harness/control/BeamFnControlClient.java

```diff
@@ -76,9 +76,11 @@ public BeamFnControlClient(
           handlers) {
     this.bufferedInstructions = new LinkedBlockingDeque<>();
     this.outboundObserver =
-        outboundObserverFactory.outboundObserverFor(
-            BeamFnControlGrpc.newStub(channelFactory.forDescriptor(apiServiceDescriptor))::control,
-            new InboundObserver());
+        SynchronizedStreamObserver.wrapping(
```

Comment:
I don't believe we need to do this since both the [DirectStreamObserver](https://github.com/apache/beam/blob/master/sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/DirectStreamObserver.java) and [BufferingStreamObserver](https://github.com/apache/beam/blob/master/sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/BufferingStreamObserver.java) are marked as thread safe and they are the only types that are returned by the HarnessStreamObserverFactories.

It might be best to update the javadoc for [OutboundObserverFactory](https://github.com/apache/beam/blob/master/sdks/java/fn-execution/src/main/java/org/apache/beam/sdk/fn/stream/OutboundObserverFactory.java) and/or [HarnessStreamObserverFactories](HarnessStreamObserverFactories.fromOptions(options);
).

Reply:
Oh, I see. Will update the javadoc in this PR. Thank you!